http/sys_metrics.go

```diff
@@ -35,12 +35,14 @@ func handleMetricsUnauthenticated(core *vault.Core) http.Handler {
 		resp := core.MetricsHelper().ResponseForFormat(format)
 
 		// Manually extract the logical response and send back the information
-		w.WriteHeader(resp.Data[logical.HTTPStatusCode].(int))
+		status := resp.Data[logical.HTTPStatusCode].(int)
 		w.Header().Set("Content-Type", resp.Data[logical.HTTPContentType].(string))
 		switch v := resp.Data[logical.HTTPRawBody].(type) {
 		case string:
+			w.WriteHeader(status)
 			w.Write([]byte(v))
 		case []byte:
+			w.WriteHeader(status)
```

Comment:
I might be missing something subtle but the proposed changes look functionally equivalent to the prior logic. A call to `w.WriteHeader` is being done in both the `string` and `[]byte` cases. Is there a reason that the outer call to `w.WriteHeader` on line 38 cannot be kept as is?

Reply:
So, once the WriteHeader is called, setting other headers would be ignored. So, in this case, the "Content-Type" will be ignored which it seems like an old bug. Another pointer is that the responseError function also sets some headers and calls the WriteHeader. So, I made the change to accommodate the two cases here.